src/pages/HostingPreferences.vue

```diff
@@ -21,35 +21,81 @@ import PrimaryLayout from 'components/PrimaryLayout.vue'
 import HAppSelectionSection from 'components/settings/hostingPreferences/HAppSelectionSection'
 import InvoicesSection from 'components/settings/hostingPreferences/InvoicesSection.vue'
 import PricesSection from 'components/settings/hostingPreferences/PricesSection.vue'
-import { reactive } from 'vue'
+import { usePreferencesStore } from '@/store/preferences'
+import { reactive, computed, onMounted} from 'vue'
 
-const invoicesSettings = reactive({
+const preferencesStore = usePreferencesStore()
+
+const localInvoicesSettings = reactive({
   frequency: {
     amount: 100,
     period: 7
   },
   due: {
     period: 7
-  }
+  },
 })
 
-const pricesSettings = reactive({
+const localPricesSettings = reactive({
   cpu: 100,
   storage: 100,
-  bandwidth: 100
+  bandwidth: 100,
 })
 
 function updateFrequency(value) {
-  invoicesSettings.frequency = value
+  localInvoicesSettings.frequency = value
 }
 
 function updateDue(value) {
-  invoicesSettings.due = value
+  localInvoicesSettings.due = value
 }
 
 function updatePrice({ prop, value }) {
-  pricesSettings[prop] = value
+  localPricesSettings[prop] = value
 }
+
+const pricesSettings = computed(() => ({
+  ...localPricesSettings,
+  ...preferencesStore.pricesSettings
+}))
+
+const invoicesSettings = computed(() => ({
+  ...localInvoicesSettings,
+  ...preferencesStore.invoicesSettings
+}))
+
+// NOTE: This can currently only be called only once - ensure all values are present before invoking
+async function setHostSettings () {
```

Comment:
@mateuszRybczonek - Likewise, I placed this fn here as a placeholder and, primarily, to underscore the point that the setting of host preference is currently a single-time event, not an optional update. I didn't implement the use of this call here, as I wanted to allow space for you to incorporate it into your more general UI patterns.